File: convert_images.js
const fs = require('fs');
const path = require('path');

function imageToBase64(imagePath) {
    try {
        const imageBuffer = fs.readFileSync(imagePath);
        const base64String = imageBuffer.toString('base64');
        return base64String;
    } catch (error) {
        console.error(`Error converting ${imagePath}:`, error.message);
        return null;
    }
}

function main() {
    const imagesDir = path.join(__dirname, 'shree_public', 'images');
    const outputFile = 'base64_images.txt';
    
    try {
        const files = fs.readdirSync(imagesDir);
        const imageFiles = files.filter(file => file.toLowerCase().endsWith('.jpg'));
        
        let output = 'Base64 encoded images for HTML:\n\n';
        
        imageFiles.forEach((imageFile, index) => {
            const imagePath = path.join(imagesDir, imageFile);
            console.log(`Converting ${imageFile}...`);
            
            const base64String = imageToBase64(imagePath);
            if (base64String) {
                output += `<!-- ${imageFile} -->\n`;
                output += `<img src="data:image/jpeg;base64,${base64String}" alt="Shree Agrawal Yaatri Grah - Room ${index + 1}" style="max-width: 100%; height: auto; border-radius: 8px;">\n\n`;
                console.log(`✓ ${imageFile} converted successfully`);
            } else {
                console.log(`✗ Failed to convert ${imageFile}`);
            }
        });
        
        fs.writeFileSync(outputFile, output);
        console.log(`\nBase64 conversion completed! Check ${outputFile}`);
        
        // Also create a simple HTML snippet
        const htmlSnippet = createHTMLSnippet(imageFiles, imagesDir);
        fs.writeFileSync('gallery_snippet.html', htmlSnippet);
        console.log('Gallery HTML snippet created: gallery_snippet.html');
        
    } catch (error) {
        console.error('Error:', error.message);
    }
}

function createHTMLSnippet(imageFiles, imagesDir) {
    let html = '<div class="gallery-grid">\n';
    
    imageFiles.forEach((imageFile, index) => {
        const imagePath = path.join(imagesDir, imageFile);
        const base64String = imageToBase64(imagePath);
        
        if (base64String) {
            html += `    <div class="gallery-item">\n`;
            html += `        <img src="data:image/jpeg;base64,${base64String}" alt="Shree Agrawal Yaatri Grah - Room ${index + 1}" style="max-width: 100%; height: auto; border-radius: 8px;" loading="lazy">\n`;
            html += `    </div>\n`;
        }
    });
    
    html += '</div>';
    return html;
}

main(); 

File: convert_images.py
import base64
import os

def image_to_base64(image_path):
    """Convert image to base64 string"""
    try:
        with open(image_path, "rb") as image_file:
            encoded_string = base64.b64encode(image_file.read()).decode('utf-8')
            return encoded_string
    except Exception as e:
        print(f"Error converting {image_path}: {e}")
        return None

def main():
    images_dir = "shree_public/images"
    output_file = "base64_images.txt"
    
    # Get all jpg files
    image_files = [f for f in os.listdir(images_dir) if f.lower().endswith('.jpg')]
    
    with open(output_file, 'w', encoding='utf-8') as f:
        f.write("Base64 encoded images:\n\n")
        
        for image_file in sorted(image_files):
            image_path = os.path.join(images_dir, image_file)
            print(f"Converting {image_file}...")
            
            base64_string = image_to_base64(image_path)
            if base64_string:
                f.write(f"// {image_file}\n")
                f.write(f"const {image_file.replace('.jpg', '_base64')} = 'data:image/jpeg;base64,{base64_string}';\n\n")
                print(f"✓ {image_file} converted successfully")
            else:
                print(f"✗ Failed to convert {image_file}")
    
    print(f"\nBase64 conversion completed! Check {output_file}")

if __name__ == "__main__":
    main() 